packages/backend/server/src/core/doc/job.ts

```diff
@@ -73,4 +81,13 @@ export class DocStorageCronJob implements OnModuleInit {
       .gauge('updates_queue_count')
       .record(await this.db.update.count());
   }
+
+  @OnEvent('user.deleted')
+  async clearUserWorkspaces(payload: EventPayload<'user.deleted'>) {
+    const workspaces = await this.permission.getOwnedWorkspaces(payload.id);
```

Comment:
permission records will be deleted when the user is deleted, so when this event is triggered, the owned workspace can no longer be query through this api

Reply:
I delete the fkey on user.id